### PR TITLE
Remove used of bytes2unicode in buildbot master

### PR DESCRIPTION
**Pull request by rodrigc**
Now that buildbot master is only supported on Python 3.5+,
we can get rid of the internal usage of `bytes2unicode()`.

The semantics of `bytes.decode()` on Python 3 are well understood and consistent, so we should just use that:
https://docs.python.org/3.5/library/stdtypes.html#bytes.decode

## Contributor Checklist:

* [X] I have updated the unit tests
* [ ] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

**Comment by p12tic** [comment]
I think this could be a good chance to gradually review places where we don't need `bytes2unicode` at all and get rid of it, or where we always will get `bytes` and don't need to check with `isinstance()`. Just replacing bytes2unicode piecemeal makes code more confusing in places and will make searching for occurrences harder later.

**Comment by p12tic** [comment]
The test failure looks to be not an instability.

File: master/buildbot/changes/bitbucket.py [hunk 1]
```diff
@@ -62,9 +61,12 @@ def __init__(self, owner, slug,
         self.lastChange = time.time()
         self.lastPoll = time.time()
         self.useTimestamps = useTimestamps
-        self.category = category if callable(
-            category) else bytes2unicode(category)
-        self.project = bytes2unicode(project)
+        self.category = category
```
Comment (sa2ajj) [inline master/buildbot/changes/bitbucket.py]: i'd rather do it in other way:
```
if isinstance(category, bytes):
    category = category.decode()
self.category = category

File: master/buildbot/changes/bitbucket.py [hunk 2]
```diff
@@ -62,9 +61,12 @@ def __init__(self, owner, slug,
...
+        if isinstance(self.category, bytes):
+            self.category = self.category.decode()
+        self.project = project
```
Comment (sa2ajj) [inline master/buildbot/changes/bitbucket.py]: ditto

File: master/buildbot/changes/github.py [hunk 3]
```diff
@@ -129,8 +128,9 @@ def reconfigService(self,
         else:
             self.pullrequest_filter = (lambda _: pullrequest_filter)
 
-        self.category = category if callable(category) else bytes2unicode(
-            category)
+        self.category = category
```
Comment (sa2ajj) [inline master/buildbot/changes/github.py]: ditto

File: master/buildbot/changes/gitpoller.py [hunk 4]
```diff
@@ -95,9 +94,12 @@ def __init__(self, repourl, branches=None, branch=None,
         self.gitbin = gitbin
         self.workdir = workdir
         self.usetimestamps = usetimestamps
-        self.category = category if callable(
-            category) else bytes2unicode(category, encoding=self.encoding)
-        self.project = bytes2unicode(project, encoding=self.encoding)
+        self.category = category
```
Comment (sa2ajj) [inline master/buildbot/changes/gitpoller.py]: ditto

File: master/buildbot/changes/gitpoller.py [hunk 5]
```diff
@@ -95,9 +94,12 @@ def __init__(self, repourl, branches=None, branch=None,
...
+        if isinstance(category, bytes):
+            self.category = self.category.decode()
+        self.project = project
```
Comment (sa2ajj) [inline master/buildbot/changes/gitpoller.py]: ditto

File: master/buildbot/changes/hgpoller.py [hunk 6]
```diff
@@ -72,8 +71,9 @@ def __init__(self, repourl, branch=None, branches=None, bookmarks=None,
         self.hgbin = hgbin
         self.workdir = workdir
         self.usetimestamps = usetimestamps
-        self.category = category if callable(
-            category) else bytes2unicode(category)
+        self.category = category
```
Comment (sa2ajj) [inline master/buildbot/changes/hgpoller.py]: ditto

File: master/buildbot/changes/p4poller.py [hunk 7]
```diff
@@ -156,7 +155,9 @@ def __init__(self, p4port=None, p4user=None, p4passwd=None,
         self.p4bin = p4bin
         self.split_file = split_file
         self.encoding = encoding
-        self.project = util.bytes2unicode(project)
+        self.project = project
```
Comment (sa2ajj) [inline master/buildbot/changes/p4poller.py]: ditto

File: master/buildbot/changes/svnpoller.py [hunk 8]
```diff
@@ -117,9 +116,12 @@ def __init__(self, repourl, split_file=None,
         self.svnbin = svnbin
         self.histmax = histmax
         self._prefix = None
-        self.category = category if callable(
-            category) else util.bytes2unicode(category)
-        self.project = util.bytes2unicode(project)
+        self.category = category
```
Comment (sa2ajj) [inline master/buildbot/changes/svnpoller.py]: ditto

File: master/buildbot/changes/svnpoller.py [hunk 9]
```diff
@@ -117,9 +116,12 @@ def __init__(self, repourl, split_file=None,
...
+        if isinstance(self.category, bytes):
+            self.category = self.category.decode()
+        self.project = project
```
Comment (sa2ajj) [inline master/buildbot/changes/svnpoller.py]: ditto

File: master/buildbot/changes/svnpoller.py [hunk 10]
```diff
@@ -396,22 +398,21 @@ def create_changes(self, new_logentries):
                         number_of_files_changed == 1 and files[0] == ''):
                     log.msg("Ignoring deletion of branch '%s'" % branch)
                 else:
+                    if isinstance(branch, bytes):
+                        branch = branch.decode()
                     chdict = dict(
```
Comment (sa2ajj) [inline master/buildbot/changes/svnpoller.py]: and an empty line before this one

File: master/buildbot/changes/svnpoller.py [hunk 11]
```diff
@@ -396,22 +398,21 @@ def create_changes(self, new_logentries):
...
                         author=author,
                         # weakly assume filenames are utf-8
-                        files=[bytes2unicode(f, 'utf-8', 'replace')
+                        files=[f.decode('utf-8', 'replace') if isinstance(f, bytes) else f
```
Comment (sa2ajj) [inline master/buildbot/changes/svnpoller.py]: maybe nice re-format as
```
files=[
    ...
]
```
?

File: master/buildbot/data/types.py [hunk 12]
```diff
@@ -129,7 +128,9 @@ class String(Instance):
     ramlType = "string"
 
     def valueFromString(self, arg):
-        val = util.bytes2unicode(arg)
+        val = arg
```
Comment (sa2ajj) [inline master/buildbot/data/types.py]: i'm not sure we need to introduce another variable here:
```
return arg.decode() if isinstance(arg, bytes) else arg
```

File: master/buildbot/data/types.py [hunk 13]
```diff
@@ -164,7 +165,9 @@ def __init__(self, len=None, **kwargs):
         self.len = len
 
     def valueFromString(self, arg):
-        val = util.bytes2unicode(arg)
+        val = arg
```
Comment (sa2ajj) [inline master/buildbot/data/types.py]: ditto

File: master/buildbot/data/types.py [hunk 14]
```diff
@@ -246,7 +249,9 @@ def validate(self, name, object):
             if not isinstance(propsrc, str):
                 yield "%s[%s] source %r is not unicode" % (name, k, propsrc)
             try:
-                json.loads(bytes2unicode(propval))
+                if isinstance(propval, bytes):
```
Comment (sa2ajj) [inline master/buildbot/data/types.py]: similar to the above case

**Pull request closed without merging**